README.md

```diff
@@ -17,6 +17,259 @@ Comming soon ;)
 
 ## Configuration
 
+### Basic Configuration
+The basic configuration includes all the minimal configuration to deploy the app successfully. Create a file named `boss.yml` in your project's root directory.
+
+##### `project_name`
+`string`
+
+The title or name of the project or application.
+```yml
+project_name: PROJECT_NAME
+```
+
+##### `project_description`
+`string`
+
+A short description about your project.
+```yml
+project_description: PROJECT_DESCRIPTION
+```
+
+##### `repository_url`
+`string`
+
+The location of repository where you or your team mates push the project.
+```yml
+repository_url: https://github.com/<username>/<project-name>
+```
+
+##### `username`
+`string`
+
+The user used to access the servers through SSH.
+```yml
+user: USERNAME
+```
+_**NOTE:** This user will be used to access all the servers mentioned in stages below._
+
+##### `port`
+`integer`
+
+The port used to access the servers through SSH.
+```yml
+port: SERVER_SSH_PORT
+```
+_**NOTE:** This port will be used to access all the servers mentioned in stages below._
+
+### Stages
+Stages refer to the configured remote servers where you would like to deploy your application. E.g. `dev`, `qa`, `uat`, `staging`, `production`.
+
+##### `stages`
+`array/list`
+
+The list of different stages of your application with their details.
+```yml
+stages:
+    dev:
+        ...
+    uat:
+        ...
+```
+
+##### `stages`.`<stage>`.`host`
```

Comment:
Instead of doing `stages.<stage>.host`, I think it would make more sense to write stage specific config as this:

You can define stages for each of the remote servers you have. For instance:
```yml
stages:
  dev:
    host: dev.your-app.com
    ...
  qa:
    host: qa.your-app.com
    ...
  ...
```
  

For each of the stages you can define the following configuration options.

##### `host`
The host name of the remote server/host.

```yml
host: test.your-app.com
```